### PR TITLE
Fixes style testing issues

### LINKED ISSUE #546: The word "aesthetic" does not have a supported spelling.
The word aesthetic will fail the style check, which will suggest the word be spelled "esthetic". But if that spelling is used, the spell checker will fail.

### PR DESCRIPTION
**Pull request by ankita240796**
closes #546 
addresses #471 

## What is included in this PR?
- Fixes spelling issues ``aesthetic`` and ``esthetic``
- Removes section label testing on doc title

### CODE REVIEW

**Comment by adammichaelwood** [comment]
@ankita240796 This has a merge conflict related to some renamed variables. Other than that it is ready to merge.

**Comment by ankita240796** [comment]
@adammichaelwood I have resolved the conflicts!

File: src/docs-style-guide.rst [hunk 1]
```diff
@@ -661,8 +661,12 @@ In these cases, you may want to use the :rst:dir:`rubric` directive.
 
       errors = []
       sym_list = ['===','---','~~~','"""','\'\'\'']
+      cnt_label = 0
```
Comment (adammichaelwood) [inline src/docs-style-guide.rst]: I'm merging. But, just for your own edification, here is what I would have done:

```
is_doc_title = True

for matchobj in re.finditer(regex,text):
    if is_doc_title:
        is_doc_title = False
        continue
```

This avoids keeping an ever-increasing integer for no reason (a very minor efficiency) and is also more explicitly a coded version of what is actually happening.

Comment (ankita240796) [reply src/docs-style-guide.rst]: Thank you for the suggestion! I will add this as a fix to improve the efficiency!

**Pull request merged**